kopf/structs/hierarchies.py

```diff
@@ -44,7 +52,7 @@ def append_owner_reference(objs, owner):
     owner = build_owner_reference(owner)
     for obj in dicts.walk(objs):
         refs = obj.setdefault('metadata', {}).setdefault('ownerReferences', [])
-        matching = [ref for ref in refs if ref['uid'] == owner['uid']]
+        matching = [ref for ref in refs if ref.get('uid') == owner.get('uid')]
```

Comment:
is this safe to keep refs with id None if the owner's uid is None? 
not sure if both owner's uid and ref's uid can be None is the same time though!

Reply:
Perhaps, `ref['uid']` is mandatory. I was replacing these things in almost automated way, didn't pay attention (previously, it would fail with a KeyError in such cases).

The only case, when this can happen so that `ref['uid']` is None/absent, is when we create both the owner & the child, and keep them in-memory, not yet posted to the API, therefore not yet having the uids — and then we try to connect them. Adding an additional 2nd owner would hit this mach `None == None`, despite the 1st & 2nd owners can have different apiVersion/kind/name.

This is quite an artificial case. I'm not even sure if it is possible to add owner references without a uid, just by a name. We can solve it in advance, but maybe as a separate PR — not related to the event-posting. And I'm not even sure how to properly reproduce this scenario.